guides-markdown/multisig-wallet.md

```diff
@@ -0,0 +1,370 @@
+myadmintoken="959f407c35f3bc4e33a643f9c0e6bbcf0c6c1f65c16bb2e20edcb3197c1fd034"
+curl ${url}/wallet/primary?token=${myadmintoken}
+```
+
+If it works, you will see the information of the wallet including it's `token`
+
+Since we have set `wallet-auth` to true in `wallet.conf`, each wallet's are seperated by it's own authorization token. 
+
+And then you can query by
+
+```bash
+curl $url/wallet/primary?token=${primarytoken}
+```
+
+### 2. Setting up hot wallet on the server.
+
+Now let's create a multisig account in this `primary` wallet.
+
+TODO:
+
+### 3. Setting up a watch-only multisig wallet
+
+Here, we will create a multisig account as we did above with one difference.
+This time, the wallet is watch-only. So no private key will be on the server.
+
+Let's assume this wallet is for managing funds for three people.
+You, Alice, and Bob.
+
+> NOTE: Strictly speaking, there is no such a thing *watch-only account*.
+> Accounts will never hold a private key no matter the wallet which it belongs to holds it's master private key or not.
+> If the wallet does, then it is able to derive account private key so the account will be *hot* .
+> In fact, this is the way how bcoin actually handles it's private key for an account, it derives every time when it's need.
+> If the wallet doesn't hold master private key (i.e. if it's watch-only), then an account derived from the wallet will be also watch-only.
+
+we can set watch-only wallet named `sharedWallet1` on bcoin node by
+
+```bash
+curl -X PUT $url/wallet/sharedWallet1 --data '
+  {"watchOnly": true, "type": "multisig", "m": 2, "n": 3 }
+  '
+
+# or by cli tool (installed with `npm i -g bclient`)
+bwallet-cli \
+  --url=${url} \
+  --network=testnet \
+  --api-key=mySecretApiKey \
+  --token=${myadmintoken} \ # Must use your admin token to create a new wallet
+  --watch-only=true \ # make it watch only
+  --type=multisig \
+  --m=2 \
+  --n=3 \
+  mkwallet sharedWallet1
+```
+
+Wallet will always have account named `default` when it's created.
+This is the multisig account we are going to use.
+
+> NOTE: this time, we specified `type` for the wallet. But strictly speaking again, wallet itself does not hold property if it is multisig or not.
+> So some options we have passed here (i.e. `type`, `m`, and `n`) are for the default account created with the wallet.
+
+If the wallet is created successfully, than you will see the access token for that wallet in the return value.
+So let's set that token into variable named `sharedWalletToken` and pass to co-signers.
+
+You must register your public key to this `default` account. Let's do it by using bcoin api.
+
+TODO: create and register key.
+
+```js
+
+```
+
+Next, cosigner Alice should create tpub too and set it to the account.
+Although it is possible that Alice can create it in the same way you did,
+lets assume that Alice uses [bitcoinjs-lib](https://github.com/bitcoinjs/bitcoinjs-lib) in her software.
+So the code will be something like ...
+
+```js
```

Comment:
While you can definitely use `bitcoinjs`, there are also utilities in `bcoin` to do everything that you would need to do here. The other guides are really good for getting up to speed on this